pkg/services/ngalert/remote/alertmanager.go

```diff
@@ -247,10 +247,45 @@ func (am *Alertmanager) TestTemplate(ctx context.Context, c apimodels.TestTempla
 }
 
 func (am *Alertmanager) StopAndWait() {
+	am.sender.Stop()
 }
 
+// TODO: change implementation, this is only useful for testing other methods
 func (am *Alertmanager) Ready() bool {
-	return false
+	readyURL := strings.TrimSuffix(am.url, "/") + "/-/ready"
+	res, err := am.httpClient.Get(readyURL)
+	if err != nil {
+		return false
+	}
+
+	defer func() {
+		if err := res.Body.Close(); err != nil {
+			am.log.Warn("Error closing response body", "err", err)
+		}
+	}()
+
+	if res.StatusCode != http.StatusOK {
+		am.log.Debug("Alertmanager readiness check unsuccessful", "statusCode", res.StatusCode)
+		return false
+	}
+
+	// Wait for active senders.
```

Comment:
We're gonna keep this part for the final implementation of `Alertmanager.Ready()`